### ISSUE TITLE
Tailscale on Apple TV has serious packet loss and dropped connections

### ISSUE THREAD

**Issue by rstoeber** [tailscale/tailscale#9736]
### What is the issue?

I configured Apple TV as a new node on my tailnet, and it basically works as expected. However, I'm running into several issues:

1) When I ping the Apple TV using the private tailnet IP address I get a lot of packet loss. For testing I set up an old Raspberry Pi as a tailnet node in the same location at the Apple TV. Pinging the private address on the Raspberry Pi has no packet loss.

2) When I try to use the Apple TV as an Exit node it generally works, but again I see poor performance and disconnects.

### Steps to reproduce

Install Tailscale on Apple TV. Try to use it. 

Observe packet loss, poor performance and disconnects when Apple TV is used as an exit node.

### Are there any recent changes that introduced the issue?

installed Tailscale on Apple TV

### OS version

tvOS 17.1

### Tailscale version

Tailscale for tvOS 1.50.1

### Bug report

no idea how to do this on Apple TV

**Comment by agottardo** [comment]
Are you using Wi-Fi or Ethernet? Did this regress upon upgrading to the tvOS 17.1 beta, or have you always seen it? Is there anything peculiar about your LAN setup? Can you reliably reproduce this on another Apple TV, if available?

I'm working on a way to file a bug report from the tvOS client, will post more updates here when available.

**Comment by rstoeber** [comment]
Andrea,

My Apple TV is connected via ethernet cable.

I didn't get Tailscale on the Apple TV until after I installed tvOS 17.1 so I'm not sure about earlier versions. But, I just got a new tvOS update (beta 3) over the weekend, so the same problem exists in today's beta update as well as the earlier version of 17.1.

But, I found something interesting today. When I ping the Apple TV local (router-assigned) address, I get exactly the same packet loss. In other words, this problem might not have anything to do with Tailscale.

I am doing this on my MacBook, connected to the same home ethernet as the Apple TV. 

When the Apple TV is awake, it works perfectly. 

When the Apple TV is asleep I consistently get bad results like this:


rstoeber@airhead2 ~ % ping 192.168.10.221
PING 192.168.10.221 (192.168.10.221): 56 data bytes
64 bytes from 192.168.10.221: icmp_seq=0 ttl=64 time=2.030 ms
64 bytes from 192.168.10.221: icmp_seq=1 ttl=64 time=1.375 ms
64 bytes from 192.168.10.221: icmp_seq=2 ttl=64 time=1.539 ms
64 bytes from 192.168.10.221: icmp_seq=3 ttl=64 time=1.436 ms
64 bytes from 192.168.10.221: icmp_seq=4 ttl=64 time=1.239 ms
64 bytes from 192.168.10.221: icmp_seq=5 ttl=64 time=1.366 ms
64 bytes from 192.168.10.221: icmp_seq=6 ttl=64 time=1.206 ms
64 bytes from 192.168.10.221: icmp_seq=7 ttl=64 time=1.374 ms
64 bytes from 192.168.10.221: icmp_seq=8 ttl=64 time=1.472 ms
64 bytes from 192.168.10.221: icmp_seq=9 ttl=64 time=1.319 ms
64 bytes from 192.168.10.221: icmp_seq=10 ttl=64 time=1.318 ms
64 bytes from 192.168.10.221: icmp_seq=11 ttl=64 time=1.264 ms
64 bytes from 192.168.10.221: icmp_seq=12 ttl=64 time=1.305 ms
Request timeout for icmp_seq 13
Request timeout for icmp_seq 14
Request timeout for icmp_seq 15
Request timeout for icmp_seq 16
Request timeout for icmp_seq 17
Request timeout for icmp_seq 18
64 bytes from 192.168.10.221: icmp_seq=19 ttl=64 time=1.913 ms
64 bytes from 192.168.10.221: icmp_seq=20 ttl=64 time=1.016 ms
64 bytes from 192.168.10.221: icmp_seq=21 ttl=64 time=1.737 ms
^C
--- 192.168.10.221 ping statistics ---
22 packets transmitted, 16 packets received, 27.3% packet loss
round-trip min/avg/max/stddev = 1.016/1.432/2.030/0.255 ms


Right now this appears to be an Apple networking issues, possibly related to tvOS 17.1 beta releases. I will keep testing and post updates if I learn anything new.

**Comment by agottardo** [comment]
Do you have your Apple TV set up as a Home Hub and assigned to a room in your home? If not, enabling that feature might help as in our testing it seems to disable some power saving optimizations.

**Comment by rstoeber** [comment]
I do not have any HomeKit devices, but under the Apple TV "AirPlay and HomeKit" settings I do have a room name set to Living Room. Is that what you mean? 

For now I have set Sleep = Never and I will test more from my office tomorrow. If I am right that this problem is related to sleeping, I should be able to use my Apple TV as a Tailscale exit node reliably tomorrow.

**Comment by bradfitz** [comment]
Sleep=Never refers to sleeping the attached TV, not sleeping the Apple TV itself, IIUC.

**Comment by rstoeber** [comment]
I'm no expert, and @bradfitz your Apple TV might be different than mine. On my Apple TV when I set SLEEP = NEVER the little light on the Apple TV stays on forever. If I set it for SLEEP after 5 minutes, the light turns off after 5 minutes. I assume that means the Apple TV itself has fallen asleep.

Also, there is a message when setting the sleep option that says "Apple TV can automatically turn off after a period of inactivity." Not sure why it says "turn off" because it's definitely not "off" even if the light is off - I can still ping it and it's still active on my tailnet.

See attached picture for the message I see about sleep / off.
![Apple-TV-sleep](https://github.com/tailscale/tailscale/assets/3082512/089cd279-6cf2-4f6c-a2f7-9db6dff81c22)

**Comment by rstoeber** [comment]
Not exactly sure what fixed the problem for me because two changes happened at once.

I upgraded to the release candidate for tvOS 17.1

Also, as suggested by @agottardo I configured the Apple TV as a Home Hub even though I have no devices to control.

Either way, I now have a perfect connection to my Apple TV. There are no more dropped packets at all, even when the Apple TV has been asleep all night. It also works perfectly as a tailnet exit node.
